server/server.go

```diff
@@ -167,7 +168,7 @@ func (serv *shipperServer) PersistedIndex(req *messages.PersistedIndexRequest, p
 	serv.logger.Debug("new subscriber for persisted index change")
 	defer serv.logger.Debug("unsubscribed from persisted index change")
 
-	persistedIndex := serv.GetPersistedIndex()
+	persistedIndex := serv.getPersistedIndex()
```

Comment:
I'm not sure returning 0 when the queue is shutting down makes sense here, when we handle this case below. The only job of the RPC here is to return the queue persistent index so when we can read it it makes sense to just return an error. The shipper is presumably going to shutdown soon anyway.


```suggestion
	persistedIndex, err := serv.publisher.PersistedIndex()
	if err != nil {
	    return status.Error(codes.Unavailable, err)
	}
```

We could just return the error directly back to the caller here

Reply:
done